RadixWallet/Features/DappInteractionFeature/Children/PersonaProofOfOwnership/PersonaProofOfOwnership.swift

```diff
@@ -0,0 +1,107 @@
+// MARK: - PersonaProofOfOwnership
+@Reducer
+struct PersonaProofOfOwnership: Sendable, FeatureReducer {
+	@ObservableState
+	struct State: Sendable, Hashable {
+		let identityAddress: IdentityAddress
+		let dappMetadata: DappMetadata
+		let challenge: DappToWalletInteractionAuthChallengeNonce
+
+		var persona: Persona?
+
+		init(
+			identityAddress: IdentityAddress,
+			dappMetadata: DappMetadata,
+			challenge: DappToWalletInteractionAuthChallengeNonce
+		) {
+			self.identityAddress = identityAddress
+			self.dappMetadata = dappMetadata
+			self.challenge = challenge
+		}
+	}
+
+	typealias Action = FeatureAction<Self>
+
+	enum ViewAction: Sendable, Equatable {
+		case task
+		case continueButtonTapped
+	}
+
+	enum InternalAction: Sendable, Equatable {
+		case setPersona(Persona)
+	}
+
+	enum DelegateAction: Sendable, Equatable {
+		case provenOwnership(Persona, SignedAuthChallenge)
+		case failedToGetPersona
+		case failedToSign
+	}
+
+	@Dependency(\.personasClient) var personasClient
+	@Dependency(\.rolaClient) var rolaClient
+	@Dependency(\.deviceFactorSourceClient) var deviceFactorSourceClient
+
+	var body: some ReducerOf<Self> {
+		Reduce(core)
+	}
+
+	func reduce(into state: inout State, viewAction: ViewAction) -> Effect<Action> {
+		switch viewAction {
+		case .task:
+			loadPersonaEffect(state: state)
+		case .continueButtonTapped:
+			signChallengeEffect(state: state)
+		}
+	}
+
+	func reduce(into state: inout State, internalAction: InternalAction) -> Effect<Action> {
+		switch internalAction {
+		case let .setPersona(persona):
+			state.persona = persona
+			return .none
+		}
+	}
+
+	private func loadPersonaEffect(state: State) -> Effect<Action> {
+		.run { send in
+			let persona = try await personasClient.getPersona(id: state.identityAddress)
+			await send(.internal(.setPersona(persona)))
+		} catch: { error, send in
+			loggerGlobal.error("Failed to get Persona to proove its ownership, \(error)")
+			await send(.delegate(.failedToGetPersona))
+		}
+	}
+
+	private func signChallengeEffect(state: State) -> Effect<Action> {
```

Comment:
I would really prefer it if this function could be merged with `gatherSignaturePayloadsEffect`

Reply:
I don't think this is something we want to do. 

For `PersonaProofOfOwnership` (`signChallengeEffect`), the logic could be extracted to some client or global function because the signature is handled by the `deviceFactorSourceClient`.

However, for `AccountsProofOfOwnership` (`gatherSignaturesEffect`), the solution takes two steps: first we gather the signatures inside the mentioned effect, and then we need to present the `Signing.View`, and perform specific logic on each of the 4 delegate calls that the `Signing` reducer can make. 

There aren't really too many things in common, asides from the creation of a `AuthenticationDataToSignForChallengeRequest` used by the `RolaClient`.

Reply:
@CyonAlexRDX in the context of MFA, for personas it would be possible to have other factor sources right?

Also, wouldn't the following the return the device factor source when signing for a persona? I mean instead of using specifically the device factor source, we can let the factorSourcesClient drive this, and cover the potential future scenarios when there will be other other factor sources that can sign for persona.
```
let signingFactors = try await factorSourcesClient.getSigningFactors(.init(
				networkID: accountsClient.getCurrentNetworkID(),
				signers: signers,
				signingPurpose: .signAuth
			))
```

Reply:
Let's sync tomorrow. I think this will be replaced by ROLASigner which @micbakos-rdx writes according to the ticket

Reply:
Updated solution and isolated the logic on `SignProofOfOwnership`